sdk/eventhub/Azure.Messaging.EventHubs/tests/Core/PartitionContextTests.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Core;
+using Azure.Messaging.EventHubs.Errors;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests
+{
+    /// <summary>
+    ///   The suite of tests for the <see cref="PartitionContext" />
+    ///   class.
+    /// </summary>
+    ///
+    [TestFixture]
+    public class PartitionContextTests
+    {
+        /// <summary>
+        ///   Verifies functionality of the constructor.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(null)]
+        [TestCase("")]
+        public void ConstructorValidatesTheEventHubName(string value)
+        {
+            Assert.That(() => new PartitionContext(value, "test"), Throws.InstanceOf<ArgumentException>(), "The constructor with no consumer should validate.");
+            Assert.That(() => new PartitionContext(value, "test", Mock.Of<TransportConsumer>()), Throws.InstanceOf<ArgumentException>(), "The constructor with no consumer should validate.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the constructor.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(null)]
+        [TestCase("")]
+        public void ConstructorValidatesThePartition(string value)
+        {
+            Assert.That(() => new PartitionContext("hub-name", value), Throws.InstanceOf<ArgumentException>(), "The constructor with no consumer should validate.");
+            Assert.That(() => new PartitionContext("hub-name", value, Mock.Of<TransportConsumer>()), Throws.InstanceOf<ArgumentException>(), "The constructor with no consumer should validate.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the constructor.
+        /// </summary>
+        ///
+        [Test]
+        public void ConstructorValidatesTheConsumer()
+        {
+            Assert.That(() => new PartitionContext("hub-name", null), Throws.ArgumentNullException);
```

Comment:
This constructor call is validating the partition id, not the consumer as intended. Should be `hub-name, partitionId, null`.

Reply:
Good catch!  Forgot to update that after changing the constructor signature.